app/qml/PositionMarker.qml

```diff
@@ -11,6 +11,42 @@ Item {
     property QgsQuick.PositionKit positionKit
     property color baseColor: InputStyle.highlightColor
     property bool withAccuracy: true
+    // animated position
+    property real animatedPosX
+    property real animatedPosY
+
+    function changePos() {
+        var posNext = positionKit.mapSettings.coordinateToScreen(positionKit.projectedPosition)
+
+        animateX.from = animatedPosX
+        animateX.to = posNext.x
+        animateX.restart()
+
+        animateY.from = animatedPosY
+        animateY.to = posNext.y
+        animateY.restart()
+    }
+
+    Component.onCompleted: {
+        animatedPosX =  positionKit.projectedPosition.x
+        animatedPosY = positionKit.projectedPosition.y
```

Comment:
shouldn't this use coordinates transformed to screen? like in the code on canvas extents change?